src/dbnode/integration/admin_session_fetch_bootstrap_blocks_metadata_test.go

```diff
@@ -0,0 +1,145 @@
+package integration
+
+import (
+	"fmt"
+	"github.com/m3db/m3/src/dbnode/client"
+	"github.com/m3db/m3/src/dbnode/integration/generate"
+	"github.com/m3db/m3/src/dbnode/storage/block"
+	"github.com/m3db/m3/src/dbnode/storage/bootstrap/result"
+	"github.com/m3db/m3/src/x/ident"
+	xtime "github.com/m3db/m3/src/x/time"
+	"github.com/stretchr/testify/require"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestAdminSessionFetchBootstrapBlocksMetadataFromPeer(t *testing.T) {
+
+	if testing.Short() {
+		t.SkipNow()
+	}
+
+	numOfActiveSeries := 1000
+	writeBatchSize := 100
+	readBatchSize := 10
+
+	testOpts := NewTestOptions(t).SetUseTChannelClientForWriting(true).SetNumShards(1).SetFetchSeriesBlocksBatchSize(readBatchSize)
+	testSetup, err := NewTestSetup(t, testOpts, nil)
+	require.NoError(t, err)
+	defer testSetup.Close()
+
+	// Start the server
+	log := testSetup.StorageOpts().InstrumentOptions().Logger()
+	require.NoError(t, testSetup.StartServer())
+
+	// Stop the server
+	defer func() {
+		require.NoError(t, testSetup.StopServer())
+		log.Debug("server is now down")
+	}()
+
+	start := testSetup.NowFn()()
+	testSetup.SetNowFn(start)
+
+	// Write test data
+	writeTestData(t, testSetup, testNamespaces[0], start, numOfActiveSeries, writeBatchSize)
+
+	testSetup.SetNowFn(testSetup.NowFn()().Add(10 * time.Minute))
+	end := testSetup.NowFn()()
+
+	// Fetch and verify metadata
+	observedSeries := getTestSetupBootstrapBlocksMetadata(t, testSetup, testNamespaces[0], start, end)
+	verifySeriesMetadata(t, numOfActiveSeries, observedSeries)
+}
+
+func writeTestData(t *testing.T, testSetup TestSetup, namespace ident.ID, start xtime.UnixNano, numOfSeries int, batchSize int) {
+	var wg sync.WaitGroup
+
+	for i := 0; i < numOfSeries; i += batchSize {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			for j := 0; j < batchSize; j++ {
+				id := fmt.Sprintf("foo_%d_%d", i, j)
+				currInput := generate.BlockConfig{IDs: []string{id}, Start: start, NumPoints: 5}
+				testData := generate.Block(currInput)
+				require.NoError(t, testSetup.WriteBatch(namespace, testData))
+			}
+		}()
+	}
+	wg.Wait()
+}
+
+func getTestSetupBootstrapBlocksMetadata(t *testing.T,
```

Comment:
nit: don't prefix go functions with get, getters are non-idiomatic in go

call this `newTestSetupBootstrapBlocksMetadata`

Reply:
Updated the function names.